packages/mdc-textfield/types.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * @license
+ * Copyright 2019 Google Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+import {MDCRipple, MDCRippleFoundation} from '@material/ripple';
+import {MDCTextFieldCharacterCounter} from './character-counter';
+import {MDCTextFieldCharacterCounterFoundation} from './character-counter/foundation';
+import {MDCTextFieldHelperText} from './helper-text';
+import {MDCTextFieldHelperTextFoundation} from './helper-text/foundation';
+import {MDCTextFieldIcon} from './icon';
+import {MDCTextFieldIconFoundation} from './icon/foundation';
+
+export interface NativeInputType {
```

Comment:
I take it this type is only needed due to the default return of `getNativeInput_` in text field's foundation... I wonder if we can get away w/o that but I suppose this might not be the PR to find that out in.

Reply:
Correct